scipy/linalg/_multimethods.py

```diff
@@ -0,0 +1,344 @@
+import functools
+import numpy as np
+from scipy._lib.uarray import Dispatchable, all_of_type, create_multimethod
+from scipy.linalg import _api
+from scipy.linalg._backend import scalar_tuple_callable_array
+
+__all__ = [
+    # solvers
+    'solve_sylvester',
+    'solve_continuous_lyapunov', 'solve_discrete_lyapunov',
+    'solve_lyapunov',
+    'solve_continuous_are', 'solve_discrete_are',
+    # decomp (eigen value problems)
+    'eig', 'eigvals', 'eigh', 'eigvalsh',
+    'eig_banded', 'eigvals_banded',
+    'eigh_tridiagonal', 'eigvalsh_tridiagonal', 'hessenberg', 'cdf2rdf',
+    # matrix functions
+    'expm', 'cosm', 'sinm', 'tanm', 'coshm', 'sinhm',
+    'tanhm', 'logm', 'funm', 'signm', 'sqrtm',
+    'expm_frechet', 'expm_cond', 'fractional_matrix_power',
+    'khatri_rao'
+]
+
+
+_create_linalg = functools.partial(
+                    create_multimethod,
+                    domain="numpy.scipy.linalg"
+                 )
+
+
+_mark_scalar_tuple_callable_array = functools.partial(
+                            Dispatchable,
+                            dispatch_type=scalar_tuple_callable_array,
+                            coercible=True
+                        )
+
+
+def _get_docs(func):
+    func.__doc__ = getattr(_api, func.__name__).__doc__
+    return func
+
+
+def _a_b_q_replacer(args, kwargs, dispatchables):
+    def self_method(a, b, q, *args, **kwargs):
+        return dispatchables + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@_create_linalg(_a_b_q_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def solve_sylvester(a, b, q):
+    return a, b, q
+
+
+def _a_q_replacer(args, kwargs, dispatchables):
```

Comment:
This and `_a_b_replacer` further down seem identical, as does `_A_E_replacer`. Why do we need multiple of these, just with a different name to match parameter names? It looks to me like this could be simplified.

Reply:
> Why do we need multiple of these, just with a different name to match parameter names?

Yes, see https://github.com/scipy/scipy/pull/14407#discussion_r669473513